### ISSUE TITLE
"Installation Failed" on Android 12 / App does not open first try

### ISSUE THREAD

**Issue by Defectrix** [CleverRaven/Cataclysm-DDA#60142]
### Describe the bug

When opening the app for the first time, the installation of the game data fails almost immediately. It seems to work correctly when I leave the process on the background, as the game menu shows up normally when I come back to it later.
The game also seems to have a problem starting after installation, as I often have to open the app multiple times until it works, due to unexpected crashes (no errors or pop-ups; just returns to homescreen).

### Steps to reproduce

1. Install the game (happens on both stable and experimental)
2. "Installation Failed" shows up in about 3 seconds; "Installing Game Data" shows underneath it when pressing Ok, which returns to homescreen
3. Eventually make it work by either trying constantly or leaving the installation process in the background
4. Leave the game and try to play again after a while and it will fail to boot immediately, making you open the app about 2-3 times
5. Clearing the app's data makes the issue reproducible again

### Expected behavior

Installation should work properly on first boot to ensure no file gets corrupted during the process, and the app should open without having to retry multiple times.

### Screenshots

https://cdn.discordapp.com/attachments/598523535169945607/1008289853030543360/Screenshot_2022-08-14-05-11-49-170_com.cleverraven.cataclysmdda.experimental.jpg (other directories also appear when retrying; doesn't seem to happen exclusively with any specific file/folder)

### Versions and configuration

- OS: Android
    - OS Version: Manufacturer: Xiaomi; Model: M1908C3JGG; Release: 12; Incremental: V13.0.2.0.SCUMIXM; 
- Game Version: 0df7d8f [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

https://filen.io/d/bb66cf6a-0031-45c4-878e-b820b35dcc84#!YwNq6O2FLbChL3RnnMSjeiksQS6JNyI8 (crash.log and debug.log)

**Comment by ZhilkinSerg** [comment]
Data installation process was not changed in many years. I assume there is something wrong on your device - lack of free disk space, permissions issue, some system process interfering with installation process, etc. In any case we can't help.

**Comment by Defectrix** [comment]
I've got plenty of storage, so I already ruled that out. I did assume it could have been permission issues, but everything seems fine on both app settings and App Manager. Now, it could be the third one, as MIUI has a long history of messing things up with their optimizations, but I just find it weird since I know there are other users with Xiaomi devices who have never mentioned this before.

Considering the installation issue has a workaround though, I really wanted to inquire about the crashes; the debug log (included in the Additional Context section) says it is a segmentation fault. It still crashes like that from time to time, same error. Is there any way to pinpoint why this could be happening? I feel like it's worth noting I've never had problems with games installing game data on this device before, neither did I ever crash anywhere else like that, so it's weird it's happening on Cataclysm.

**Comment by ZhilkinSerg** [comment]
I don't think we write anything to `crash.log` during installation process (or anywhere else during launch menu). This file only contains crash information after data is successfully installed and game itself is running

**Comment by Defectrix** [comment]
Well, it's fine. Thanks for trying to help me anyway! I will close the issue now.

**Comment by dtrail** [comment]
I have the same issue, but only with the newer releases from github. The play store version works fine, which is weird, is the process itself hasn't changed. I was suspecting storage isolation first, but even circumventing that didn't help.

Device: Xiaomi Mi10T Pro (apollo).

Btw, there's no workaround working for me.

Edit: I got it to work. I had to uninstall and reinstall the app via "Install with options", marking the app as a complete package and automatically assigning all permissions. Somehow, if installed the normal way, se permissions cannot be granted properly. Since I'm on Android 14 I also applied the NoStorageRestrict tweak via LSPosed.
